### PR TITLE
ENH: Phasic Policy Gradient

### PR DESCRIPTION
**Pull request by AntoineTheb**

### CODE REVIEW

File: article/_posts/2020-09-25-Phasic-Policy-Gradient.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+To try and have the best of both worlds, the authors propose the _Phasic Policy Gradient_ algorithm, where feature sharing is kept but the two networks training are decoupled.
+
+# Methods
+
+![](/article/images/phasic-pg/fig1.jpeg)
+
+To decouple training while also enabling feature sharing, a novel actor-critic architecture is proposed where the actor and critic use separate networks, while also adding a critic head to the actor network. Figure 1 highlights the proposed architecture. 
+
+To train the networks, the learning process is split into two phases: The policy phase and the auxiliary phase. During the policy phase, the actor is trained using the standard PPO[^1] clipped lost:
```
Comment (jhlegarreta) [inline article/_posts/2020-09-25-Phasic-Policy-Gradient.md:32]: *lost* : *loss*

File: article/_posts/2020-09-25-Phasic-Policy-Gradient.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+> We find that training with additional auxiliary epochs is generally beneficial, with performance tapering off around 6 auxiliary epochs.
+
+![](/article/images/phasic-pg/res4.jpeg)
+
+> It is clear that performance suffers when we perform auxiliary phases too frequently. We conjecture that each auxiliary phase interferes with policy optimization and that performing frequent auxiliary phases exacerbates this effect.
+
+## Single network vs. multiple networks
+
+![](/article/images/phasic-pg/res5.jpeg)
```
Comment (jhlegarreta) [inline article/_posts/2020-09-25-Phasic-Policy-Gradient.md:84]: How is the `when necessary` part determined?

**Review by jhlegarreta — COMMENTED** [review]
Some comments in-line.

File: article/_posts/2020-09-25-Phasic-Policy-Gradient.md (hunk 2 above)
Comment (AntoineTheb) [reply article/_posts/2020-09-25-Phasic-Policy-Gradient.md:84]: Because the algorithm is modified to only use one network instead of two, loss functions that were ran on the critic network (for example) that wouldn't influence the actor network now would influence it if the gradients were not detached. I'll clarify.

**Review by jhlegarreta — APPROVED** [review]

**Review by jodoin — APPROVED** [review]

File: article/_posts/2020-09-25-Phasic-Policy-Gradient.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+---
+
+
+# Highlights
+
+- More sample-efficient actor-critic reinforcement learning algorithm 
+
+# Introduction
+
+Most policy gradient (PG) reinforcement learning (RL) algorithms use the Actor-Critic (AC) framework. In this context, two neural networks are trained in parallel: one being the Actor, aka the policy, which will output actions that are applied to the environment, the second network being the Critic, which will infer the value of the actions output by the actor. By using the training signal provided by the critic, the actor can learn more efficiently.
```
[the remark this answers is not in the document]
Comment (AntoineTheb) [reply article/_posts/2020-09-25-Phasic-Policy-Gradient.md]: Reworded
